meta-mender-toradex-nxp/classes/mender_tezimg.bbclass

```diff
@@ -1,24 +1,31 @@
-# Deploy image metadata before image-generation
-do_image[depends] += "mender-toradex-metadata:do_deploy virtual/bootloader:do_deploy"
-
-def mender_rootfs_get_size(d):
-    import subprocess
-
-    # Calculate size of rootfs in kilobytes.
-    output = subprocess.check_output(["du", "-ks",
-                                      d.getVar("IMAGE_ROOTFS", True)])
-    return int(output.split()[0])
+do_image_mender_tezimg[depends] += "tezi-metadata:do_deploy virtual/bootloader:do_deploy"
 
+# The ${IMAGE_NAME} variable extends to:
+#       ${IMAGE_BASENAME}-${MACHINE}-${DATETIME}
+# Which changes over the course of time, so we store the actual name
+# in a dedicated variable.
+def mender_get_image_name(image_type, d):
+    import re
+    image_name_base = d.getVar("IMAGE_BASENAME", True) + "-" + \
+                      d.getVar("MACHINE", True)
+    match = re.search("%s-[0-9]{14}\.%s" % \
+                      (image_name_base, image_type), \
+                      " ".join(os.listdir(d.getVar("IMGDEPLOYDIR", True))))
+    if match != None:
+        return match.group(0)
+    return ""
```

Comment:
I think we should drop this. Using `IMAGE_NAME` should be just fine, as long as there is a dependency on the `ubimg` (which there already is below).

There is one bug in Yocto that if you build the image, and then build it again with a change that *only* affects the `tezimg`, then it won't be able to find the `ubimg`, since indeed `IMAGE_NAME` is wrong (it will refer to the previous build, which has now been moved out of the build directory). But this doesn't happen very often, and the workaround is easy (delete `tmp` or call `bitbake -fc rootfs <image_name>` first). I think it's better to solve this upstream in Yocto than to have this (quite hacky) workaround.